### PR TITLE
Big internal refactor of sim / freertos related things

### LINKED ISSUE #169: Pathway for ChibiOS Adoption
Several targets are still using FreeRTOS. We should come up with a plan to move these onto ChibiOS. Additionally we need to decide if F1 will move to Chibi. We have noted increased RAM usage in some tasks with Chibi (although overall RAM usage seems to be low at least on F4 targets). I think a good approach here would be to convert either Naze32 or CC3D and compare real changes in resource usage and then make a decision. If we do convert everything to Chibi we could then begin replacing quite a few of the ST peripheral library drivers with ChibiOS HAL drivers. I am happy to do the DiscoveryF4 conversion ASAP leaving just F1 targets.

Targets awaiting transition to ChibiOS:
- [x] ~~coptercontrol~~
- [ ] pipxtreme
- [x] ~~naze32~~
- [x] revomini (pending #166)

### LINKED ISSUE #1230: Simulation altitude goes NaN
- Seen in version (release version or githash): Release-20160720-260-g51bfc90
- On flight controller (if applicable): simulator

### PR DESCRIPTION
**Pull request by mlyle**
Open for transparency / "dibs":

* [x] Move sim to direct posix threading / scheduling primitives
* [x] Move pipxtreme from freertos to ChibiOS
* [x] Remove freertos
* [x] Begin cleanup of freertos-related code
* [x] ~~Try to identify places that should have the chibios epilogue (uart? usb?) to trigger resched~~ defer to #1650 
* [x] ~~Anomaly: Some tasks on pi seem to not be realtime-class?~~
* [x] Need to somehow ensure precedence of Pi command line args
* [x] Anomaly: make sure pipx still works good

Fixes #1230 
Fixes #169

### CODE REVIEW

File: flight/Libraries/taskmonitor.c [hunk 1]
```diff
@@ -46,7 +46,7 @@ int32_t TaskMonitorInitialize(void)
 {
 	lock = PIOS_Mutex_Create();
 	PIOS_Assert(lock != NULL);
-	memset(handles, 0, sizeof(struct pios_thread) * TASKINFO_RUNNING_NUMELEM);
+	memset(handles, 0, sizeof(struct pios_thread *) * TASKINFO_RUNNING_NUMELEM);
```
Comment (tracernz) [inline flight/Libraries/taskmonitor.c:49]: Eeek.

**Comment by mlyle** [comment]
At this point, works as a data radio (!) ... have some degree of anomaly / blips in session (not clear if regression, baseline suck, or misconfig).

File: flight/Libraries/taskmonitor.c (hunk 1 above)
Comment (mlyle) [reply flight/Libraries/taskmonitor.c:49]: :) one advantage of opaquifying things-- brought this to light.

**Comment by mlyle** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
Successful 2.5 minute attitude-mode flight test on rpi2.

**Comment by mlyle** [comment]
PipX no worse than before.  Needs help, though.  (Bugs already open).

**Comment by mlyle** [comment]
~~BUG:  Module start races with module initialization on Posix.~~

~~On single-threaded stuff this was prevented before by priorities (systemmod is not highest prio).  However, on Posix, SystemMod can get to the point where it's starting all the modules before the init task has completed initializing them.~~

fixed

**Comment by dustin** [comment]
I flew the sim around a bit tonight.  Initially in quadcopter mode, and then in fixed wing.  I did a few test loiters and RTHs while making some config changes to try to get it to not make such big scary arcs.  :)

Saw no issues overall.

dronin_gcs

**Comment by tracernz** [comment]
jenkins, test this please

**Comment by mlyle** [comment]
rebased

**Pull request merged**